src/Events/Category_Colors/Migration/Pre_Processor.php

```diff
@@ -0,0 +1,218 @@
+		];
+
+		// Store processed data in the database.
+		$this->update_migration_data( $migration_data );
+
+		$this->update_migration_status( Status::$preprocess_completed );
+
+		/**
+		 * Fires after the preprocessor completes.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string, mixed> $migration_data The processed migration data.
+		 * @param bool                 $success        Whether processing was successful.
+		 */
+		do_action( 'tec_events_category_colors_migration_preprocessor_end', $migration_data, true );
+
+		$this->log_message( 'info', 'Preprocessing complete. Migration data prepared.', $migration_data, 'Pre_Processor' );
+
+		$this->log_elapsed_time( 'Preprocessing', $start_time );
+
+		return true;
+	}
+
+	/**
+	 * Extracts category-related values from processed settings and removes them.
+	 *
+	 * @since TBD
+	 * @return array<int, array<string, mixed>> Processed category data structured by category ID.
+	 */
+	protected function get_category_values(): array {
+		$categories        = [];
+		$filtered_settings = $this->processed_settings;
+
+		foreach ( $this->processed_settings['terms'] ?? [] as $term_id => [$slug, $name] ) {
+			foreach ( array_keys( $filtered_settings ) as $key ) {
+				if ( strpos( $key, $slug . '-' ) === 0 || strpos( $key, $slug . '_' ) === 0 ) {
+					$field_name = str_replace( [ $slug . '-', $slug . '_' ], '', $key );
+					$mapped_key = $this->get_mapped_meta_key( $field_name );
+
+					if ( null !== $mapped_key ) {
+						$meta_key = Config::$meta_key_prefix . $mapped_key;
+						$value    = $filtered_settings[ $key ];
+						$value    = ( 'no_color' === $value ) ? '' : $value;
+						// Store processed setting under category.
+						$categories[ $term_id ][ $meta_key ] = $value;
+
+						// Remove from copied array.
+						unset( $filtered_settings[ $key ] );
+					}
+				}
+			}
+
+			// Store the term_id reference itself.
+			$categories[ $term_id ]['taxonomy_id'] = $term_id;
+		}
+
+		// Replace the original array with the filtered one.
+		$this->processed_settings = $filtered_settings;
+
+		return $categories;
+	}
+
+	/**
+	 * Extracts and maps settings while applying validation rules.
+	 *
+	 * @since TBD
+	 * @return array<string, mixed> Processed settings with proper validation.
+	 */
+	protected function get_settings_values(): array {
+		$mapped_settings = [];
+
+		foreach ( Config::$settings_mapping as $old_key => $mapping ) {
+			if ( ! $mapping['import'] ) {
+				continue;
+			}
+			// Skip if the key doesn't exist in the processed settings.
+			if ( ! array_key_exists( $old_key, $this->processed_settings ) ) {
+				continue;
+			}
+
+			$value   = $this->processed_settings[ $old_key ];
+			$new_key = $mapping['mapped_key'];
+
+			// Apply validation rules.
+			switch ( $mapping['validation'] ) {
+				case 'boolean':
+					$value = filter_var( $value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE );
+					if ( is_null( $value ) ) {
+						$value = false;
+					}
```

Comment:
Returns false on failure, which is the same as forcing it to return null and then converting the null to false...without the extra steps:
```suggestion
					$value = filter_var( $value, FILTER_VALIDATE_BOOLEAN );
					
```